doc/conduit.rst

```diff
@@ -1,14 +1,239 @@
+.. _heph.conduit:
 
 #######
 Conduit
 #######
 
-Examples
+In order to write complex applications within the context of robotics, a mechanism which allows us to connect different components between different steps is to be developed. This shall cover handling and forwarding of sensor data toward closing feedback loops. While a pub-sub model using topics is the de-facto goto solution in most robotic platforms, it also introduces a big source of nondeterminism and challenges regarding testability as well as programmability. In addition, a big source of complexity comes from handling different timing of different inputs as well as rate limiting the publishing of data.
+
+.. todo:: Related Work
+
+ - Relation to ROS
+ - Relation to Zenoh
+ - Relation to Zenoh Flow
+
+*********
+Objective
+*********
+
+
+The primary goal of this project is to provide a lean robotics platform to provide a modular and testable framework for robotics development.
+
+The problem domain can be summarized with building a pipeline through which data can flow asynchronously. This includes abstractions to define, create, and working with these graphs.
+
+.. _heph.conduit.concepts:
+
+********
+Concepts
+********
+
+The core idea is centered around a defining a step requiring a given set of inputs and producing an set of outputs. Connecting different steps together is the cornerstone of `Conduit` forming a graph of connecting nodes. As a result of this, we define the following requirements from which we derive these core concepts:
+
+- **Inputs**: An input is what defines *when* a step function should get scheduled for execution.
+- **Outputs**: The output is publishing the result of the execution (the *what*).
+- **Stepper**: Steppers define *how* the inputs get consumed and the outputs produced.
+- **Graph Description**: The *why* is handled by describing the connections and hierarchy for our platform.
+
+
+Putting these concepts together will result in a **Conduit Graph** describing the hierarchical structure of **Nodes**. This graph can then be used to derive a directed execution graph which is getting executed based on the defined policies and connections.
+
+Following is the definition of these core concepts and their API. Implementations of these concepts can be found in :ref:`heph.conduit.usage`.
+
+.. note:: While the primary focus of conduit is to provide an abstraction for building a computational graph, Inputs and Outputs can be used outside of graphs as well as building blocks to interface with external sources.
+
+.. _heph.conduit.concepts.inputs:
+
+Inputs
+======
+
+As an Input to a step function is a very vague statement, the most basic requirement is the ability to signal a result. As such, we define the following interface:
```

Comment:
"As an Input to a step function is a very vague statement" not sure what that actually means. Can you be more clear here? Which properties are vague?